src/smartpeak/include/SmartPeak/ui/Widget.h

```diff
@@ -200,6 +204,26 @@ namespace SmartPeak
     std::vector<std::tuple<size_t, size_t>> selected_cells_;
   };
 
+  struct FeaturesTableWidget : public GenericTableWidget, public IFeaturesObserver
+  {
+    FeaturesTableWidget(const std::string& table_id,
+      const std::string title = "",
+      FeaturesObservable* observable = nullptr)
+    : GenericTableWidget(table_id, title)
+    {
+      if (observable) observable->addFeaturesObserver(this);
+    };
+
+    /**
+      IFeaturesObserver
+    */
+    virtual void onFeaturesUpdated() override
+    {
+      table_data_.clear();
+      data_changed_ = false;
```

Comment:
not sure to follow why we have to set data_changed_ to false here.

Reply:
it doesn't cause the table to be re-updated constantly